pkg/master/master.go

```diff
@@ -242,6 +242,10 @@ type Master struct {
 
 	// storage for third party objects
 	thirdPartyStorage storage.Interface
+	// map from api path to storage for those objects
+	thirdPartyResources map[string]*thirdpartyresourcedataetcd.REST
+	// protects the map
+	thirdPartyResourcesLock sync.RWMutex
```

Comment:
I guess it's a big change-- but I would prefer if the third party stuff was a separate object (and in a separate file) from the master.